include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -75,125 +92,269 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
         int m_nknots;
 
     public:
+        /// @brief The type of the discrete dimension representing the B-splines.
         using discrete_dimension_type = NonUniformBSplines;
 
+        /// @brief The type of a DiscreteDomain whose elements identify the B-splines.
         using discrete_domain_type = DiscreteDomain<DDim>;
 
+        /// @brief The type of a DiscreteElement identifying a B-spline.
         using discrete_element_type = DiscreteElement<DDim>;
 
+        /// @brief The type of a DiscreteVector representing an "index displacement" between two B-splines.
         using discrete_vector_type = DiscreteVector<DDim>;
 
         Impl() = default;
 
-        template <class OriginMemorySpace>
-        explicit Impl(Impl<DDim, OriginMemorySpace> const& impl)
-            : m_domain(impl.m_domain)
-            , m_nknots(impl.m_nknots)
-        {
-        }
-
-        /// @brief Construct a `Impl` using a brace-list, i.e. `Impl bsplines({0., 1.})`
+        /** @brief Constructs an Impl using a brace-list, i.e. `Impl bsplines({0., 1.})`
+         *
+         * The brace-list is usually the iterable list of knots. However, only the first and last
+         * indexes of the elements of the lists are effectively used by the constructor to build the Impl.
+         */
         explicit Impl(std::initializer_list<ddc::Coordinate<Tag>> breaks)
             : Impl(breaks.begin(), breaks.end())
         {
         }
 
-        /// @brief Construct a `Impl` using a C++20 "common range".
+        /** @brief Constructs an Impl using a std::vector.
+         *
+         * The std::vector is usually the iterable list of knots. However, only the first and last
+         * indexes of the elements of the lists are effectively used by the constructor to build the Impl.
+         */
         explicit Impl(std::vector<ddc::Coordinate<Tag>> const& breaks)
             : Impl(breaks.begin(), breaks.end())
         {
         }
 
-        /// @brief Construct a `Impl` using a pair of iterators.
+        /** @brief Construct a `Impl` using a pair of iterators.
+         *
+         * The (lower and upper) iterators are used to build the DiscreteDomain indexing the B-splines.
+         */
```

Comment:
See e.g. https://cplusplus.com/reference/iterator/ for a description of iterators. I'm not sure it is relevant to talk about a "lower iterator" or "upper iterator", I'm also not clear what an "upper iterator" is. Given your comments above I think you think it is like a `ubound` in Fortran but this is not the case.

This constructor builds the knots from which the B-splines are constructed from the break points. This constructor should specify which method is used to do this. It does not build the DiscreteDomain indexing the B-splines. It builds the DiscreteDomain which describes the knots

Reply:
@blegouix I disagree that this comment has been handled:

>This constructor builds the knots from which the B-splines are constructed from the break points. This constructor should specify which method is used to do this.

Reply:
Here the first degree knots are set equal to the first break point and the last degree knots are set equal to the last break point. This is not the only solution possible (a different method is used in the uniform case)

Reply:
Done